multimod/internal/diff/diff.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+
+	"go.opentelemetry.io/build-tools/multimod/internal/common"
+)
+
+// check
+
+// normalizeVersion ensures the version is prefixed with a `v`.
+func normalizeVersion(ver string) string {
```

Comment:
Why are we doing it?  I find it YAGNI and also it makes the tool not flexible. What if someone wants to compare with some custom tag.

Reply:
This is here because the missing `v` prefix in the version was causing problems in the past in the collector repo. So much so that we ended up implementing this logic in the Makefile. Since this tool was supposed to replace that functionality, i decided to implement it here.

I suppose this could be an additional flag passed into the diff command to allow others to avoid the code if they didn't need it?

Reply:
Adding a comment which will tell why it is added is good enough for me 😉